src/index/TextMetaData.cpp

```diff
@@ -27,19 +28,17 @@ const TextBlockMetaData& TextMetaData::getBlockInfoByWordRange(
   auto upperIt = std::lower_bound(_blockUpperBoundWordIds.begin(),
                                   _blockUpperBoundWordIds.end(), upper);
 
-  if (upper > *it) {
-    AD_THROW(
-        "No words found for the given prefix. This usually means that the "
-        "prefix is smaller than the configured minimum prefix size. This range "
-        "spans over " +
-        std::to_string(upperIt - it) + " blocks");
-  }
-
   // Use the info to retrieve an index.
+  vector<std::reference_wrapper<const TextBlockMetaData>> output;
   size_t index = static_cast<size_t>(it - _blockUpperBoundWordIds.begin());
-  assert(lower <= _blocks[index]._lastWordId);
   assert(lower >= _blocks[index]._firstWordId);
-  return _blocks[index];
+  while (it <= upperIt && it != _blockUpperBoundWordIds.end()) {
+    index = static_cast<size_t>(it - _blockUpperBoundWordIds.begin());
```

Comment:
If you above directly convert `it` and `upperIt` to two `size_t`s, then the code here vecomes much simpler. In particular
`ql::ranges::copy(ql::views::iota(startIndex, endIndex) | ql::views::transform(indexToBlock), std::back_inserter(output));` 
or something similar.